### PR TITLE
csi: update sidecars to latest release

### PR DESCRIPTION
**Pull request by rewantsoni**
updating all the CSI sidecars to the latest release.

Ref: https://github.com/rook/rook/pull/13846/files

### CODE REVIEW

**Comment by rewantsoni** [comment]
```
[rewantsoni ] oc get deployment csi-rbdplugin-provisioner -oyaml | grep image:
        image: registry.k8s.io/sig-storage/csi-provisioner:v4.0.0
        image: registry.k8s.io/sig-storage/csi-attacher:v4.5.0
        image: registry.k8s.io/sig-storage/csi-resizer:v1.10.0
        image: registry.k8s.io/sig-storage/csi-snapshotter:v7.0.1
        image: quay.io/csiaddons/k8s-sidecar:v0.8.0
        image: quay.io/cephcsi/cephcsi:v3.10.2

[rewantsoni ] oc get daemonset csi-cephfsplugin -oyaml | grep image:
        image: registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.10.0
        image: quay.io/cephcsi/cephcsi:v3.10.2

[rewantsoni ] oc get po | grep csi
csi-addons-controller-manager-8b866489-wvjm9              2/2     Running            0               5m
csi-cephfsplugin-7cplq                                    2/2     Running            1 (3m43s ago)   4m18s
csi-cephfsplugin-bmb4t                                    2/2     Running            1 (3m42s ago)   4m18s
csi-cephfsplugin-d86sm                                    2/2     Running            1 (3m41s ago)   4m18s
csi-cephfsplugin-gw59z                                    2/2     Running            1 (3m43s ago)   4m18s
csi-cephfsplugin-hwvfc                                    2/2     Running            1 (3m43s ago)   4m18s
csi-cephfsplugin-provisioner-648594c564-6x5xj             5/5     Running            1 (3m42s ago)   4m18s
csi-cephfsplugin-provisioner-648594c564-ngdxk             5/5     Running            1 (3m42s ago)   4m18s
csi-cephfsplugin-qdbcd                                    2/2     Running            1 (3m42s ago)   4m18s
csi-rbdplugin-4p25g                                       3/3     Running            1 (3m43s ago)   4m18s
csi-rbdplugin-59kxb                                       3/3     Running            1 (3m44s ago)   4m18s
csi-rbdplugin-provisioner-66c4d85dd8-4zcl8                6/6     Running            1 (3m42s ago)   4m18s
csi-rbdplugin-provisioner-66c4d85dd8-gwxhf                6/6     Running            2 (3m39s ago)   4m18s
csi-rbdplugin-pxnpd                                       3/3     Running            1 (3m43s ago)   4m18s
csi-rbdplugin-rsqfx                                       3/3     Running            1 (3m44s ago)   4m18s
csi-rbdplugin-x6mmt                                       3/3     Running            1 (3m44s ago)   4m18s
csi-rbdplugin-xhqpx                                       3/3     Running            1 (3m43s ago)   4m18s
```

File: config/manager/csi-images.yaml [hunk 1]
```diff
@@ -1,20 +1,20 @@
 ---
```
Comment (leelavg) [inline config/manager/csi-images.yaml:1]: it'd be better to add new stanza for version: v4.16

**Comment by rewantsoni** [comment]
> > Pods are Restarting because of context deadline exceeded:
> 
> * should that be fixed or only a remark? `csi.sock` doesn't appear instantaneously and I believe there'll be retry mechanism.

The csi-driver-registrar is not able to connect the csi.sock the container is not running, the image for csi-addons is ~1Gi and it takes more time to fetch, hence we see this error.
csi-driver-registrar starts at 7:16:30 where as the sock is available only at 7:17:02

```
[rewantsoni ] oc logs csi-rbdplugin-5crrm --previous
Defaulted container "csi-driver-registrar" out of: csi-driver-registrar, csi-rbdplugin, csi-addons
I0304 07:16:30.835041   69981 main.go:135] Version: v2.10.0
I0304 07:16:30.835111   69981 main.go:136] Running node-driver-registrar in mode=
I0304 07:16:30.835117   69981 main.go:157] Attempting to open a gRPC connection with: "unix:///csi/csi.sock"
I0304 07:16:30.835134   69981 connection.go:215] Connecting to unix:///csi/csi.sock
W0304 07:16:40.835863   69981 connection.go:234] Still connecting to unix:///csi/csi.sock
W0304 07:16:50.835476   69981 connection.go:234] Still connecting to unix:///csi/csi.sock
W0304 07:17:00.835451   69981 connection.go:234] Still connecting to unix:///csi/csi.sock

[rewantsoni ] oc logs csi-rbdplugin-5crrm --previous -ccsi-rbdplugin
Error from server (BadRequest): previous terminated container "csi-rbdplugin" in pod "csi-rbdplugin-5crrm" not found

[rewantsoni ] oc logs csi-rbdplugin-5crrm -ccsi-rbdplugin 
I0304 07:16:58.234658   70689 cephcsi.go:191] Driver version: v3.10.2 and Git version: 2339301c18b875d2c52145935c92175fc20a5720
I0304 07:16:58.234913   70689 cephcsi.go:268] Initial PID limit is set to 201866
I0304 07:16:58.234975   70689 cephcsi.go:274] Reconfigured PID limit to -1 (max)
I0304 07:16:58.235178   70689 cephcsi.go:223] Starting driver type: rbd with name: openshift-storage.rbd.csi.ceph.com
I0304 07:16:58.634117   70689 mount_linux.go:282] Detected umount with safe 'not mounted' behavior
I0304 07:17:01.537257   70689 rbd_attach.go:230] nbd module loaded
I0304 07:17:01.537387   70689 rbd_attach.go:244] kernel version "5.14.0-284.54.1.el9_2.x86_64" supports cookie feature
I0304 07:17:02.934470   70689 rbd_attach.go:260] rbd-nbd tool supports cookie feature
I0304 07:17:02.934836   70689 server.go:114] listening for CSI-Addons requests on address: &net.UnixAddr{Name:"/csi/csi-addons.sock", Net:"unix"}
I0304 07:17:02.934983   70689 server.go:113] Listening for connections on address: &net.UnixAddr{Name:"//csi/csi.sock", Net:"unix"}
```

File: config/manager/csi-images.yaml (hunk 1 above)
Comment (rewantsoni) [reply config/manager/csi-images.yaml:1]: Should we add stanzas for 4.14, 4.15, and 4.16 as we are supporting these 3 versions, and remove 4.11 and 4.12?

Comment (leelavg) [reply config/manager/csi-images.yaml:1]: ok, you can remove everything below and including 4.13, add 4.14, 4.15, 4.16 stanzas it'd be good if we can refer to sidecars used in those timeframes (stretch but not mandatory).

**Review by leelavg — APPROVED** [review]

**Pull request merged**